microcosm_flask/metrics_classifier.py

```diff
@@ -14,8 +14,17 @@ class StatusCodeClassifier(Classifier):
     """
     def label_result(self, result):
         _, status_code, _ = parse_response(result)
-        return str(status_code)
+        return self.normalize_status_code(status_code)
 
     def label_error(self, error):
         status_code = extract_status_code(error)
-        return str(status_code)
+        return self.normalize_status_code(status_code)
+
+    # Return only the first digit of the status code followed by xx as a string
+    # Example: 403 becomes "4xx"
+    # This will reduce the cardinality of metrics collected which will reduce cost
+    # For detailed status code errors we have logs
+    def normalize_status_code(self, status_code: int) -> str:
+        if isinstance(status_code, int):
+            return str(status_code)[:1] + "xx"
```

Comment:
If you just need the first character, why not `[0]`?

Reply:
Fair enough

Reply:
I mean, argument being that this method is technically safer, as you won't IndexError out. But I imagine you could also make that check earlier to see if the input results in an empty string or not.